lib/rspec/mocks/any_instance/expect_chain_chain.rb

```diff
@@ -24,6 +24,7 @@ def create_message_expectation_on(instance)
 
         def invocation_order
           @invocation_order ||= {
+            :with => [nil],
             :and_return => [nil],
             :and_raise => [nil],
             :and_yield => [nil]
```

Comment:
While we're in here, are there any other methods like `with` that is supported by `expect(...).to receive_message_chain` but not `expect_any_instance_of(...).to receive_message_chain`?